Sources/PerformanceMonitor/PerformanceMonitor.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+typealias MemoryUsage = (used: UInt64, total: UInt64)
+
+final class PerformanceMonitor {
+
+    typealias MonitoringHandler = (MonitoringInfo) -> Void
+
+    struct MonitoringInfo {
+        let memoryUsage: MemoryUsage
+    }
+
+    var monitoringHandler: MonitoringHandler?
+
+    private lazy var displayLink: CADisplayLink = .init(target: self, selector: #selector(step))
+
+    private var isMonitoring: Bool = false
+
+    // MARK: - Lifecycle
+
+    deinit {
+        displayLink.invalidate()
+    }
+
+    func startMonitoring() {
+        guard isMonitoring == false else {
+            return
+        }
+        displayLink.add(to: .current, forMode: .common)
+        isMonitoring = true
+    }
+
+    func stopMonitoring() {
+        guard isMonitoring else {
+            return
+        }
+        displayLink.invalidate()
```

Comment:
`CADisplayLink` will never be triggered after `invalidate` call, so you'll need to reinitialize it in `startMonitoring`